xonsh/xonfig.py

```diff
@@ -353,6 +353,7 @@ def _info(ns):
     data.extend([
         ('Python', '{}.{}.{}'.format(*PYTHON_VERSION_INFO)),
         ('PLY', ply.__version__),
+        ('history backend', env.get('XONSH_HISTORY_BACKEND', 'json')),
```

Comment:
`$XONSH_HISTORY_BACKEND` should be added as a full-fledged environment variable in environ.py. That way you will only have to set the default in one place.